src/views/InventoryDetail.vue

```diff
@@ -0,0 +1,443 @@
+    })
+  },
+  data() {
+    return {
+      facilityId: (this as any)?.stock?.items[0]?.facilityId,
+      facilities: [] as any,
+      queryString: "",
+      searchedProduct: {} as any,
+      isParentProductUpdated: false,
+      isPOUploadedSuccessfully: false,
+      facilityLocations: {}
+    }
+  },
+  ionViewDidEnter(){
+    this.fetchFacilities();
+  },
+  async beforeRouteLeave(to, from) {
+    let canLeave = false;
+    const alert = await alertController.create({
+      header: this.$t("Leave page"),
+      message: this.$t("Any edits made to this CSV will be lost."),
+      buttons: [
+        {
+          text: this.$t("STAY"),
+          handler: () => {
+            canLeave = false;
+          },
+        },
+        {
+          text: this.$t("LEAVE"),
+          handler: () => {
+            canLeave = true;
+          },
+        },
+      ],
+    });
+    if(!this.isPOUploadedSuccessfully){
+      alert.present();
+      await alert.onDidDismiss();
+      this.isPOUploadedSuccessfully = false;
+      return canLeave;
+    }
+  },
+  
+  methods: {
+    setFacilityLocation(ev: CustomEvent, product: any){
+      this.stock.items.map((item: any) => { 
+        if(item.pseudoId === product.pseudoId){
+          item.locationSeqId = ev.detail.value;
+        }
+      });
+      this.store.dispatch('stock/updatedStockListItems', this.stock.items);
+    },
+    async listMissingSkus() {
+      const missingSkuModal = await modalController.create({
+        component: MissingSkuModal,
+        componentProps: { 'unidentifiedProductItems': this.stock.unidentifiedProductItems }
+      });
+      return missingSkuModal.present();
+    }, 
+    searchProduct(sku: any) {
+      const product = this.getProduct(sku);
+      this.searchedProduct = this.stock.items.find((item: any) => {
+        return item.pseudoId === product.pseudoId;
+      })
+    },
+    async save(){
+      const uploadData = this.stock.items.filter((item: any) => {
+        return item.isSelected;
+      }).map((item: any) => {
+        return {
+          "facilityId": item.facilityId,
+          "externalFacilityId": item.externalFacilityId,
+          "idValue": item.productSKU,
+          "idType": "SKU",
+          "locationSeqId": item.locationSeqId,
+          "availableQty": item.quantity
+        };
+      })
+      const fileName = this.fileName.replace(".csv", ".json");
+      const params = {
+        "configId": "RESET_INVENTORY"
+      }
+      const alert = await alertController.create({
+        header: this.$t("Reset inventory"),
+        message: this.$t("Make sure all the data you have entered is correct."),
+        buttons: [
+            {
+              text: this.$t("Cancel"),
+              role: 'cancel',
+            },
+            {
+              text: this.$t("Upload"),
+              handler: () => {
+                const response = UploadService.uploadJsonFile(UploadService.prepareUploadJsonPayload({
```

Comment:
response is not used, so I think we can remove this variable.

Reply:
Removed unused variable.